backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -280,5 +264,104 @@ bool ValidateTableKeys::preorder(const IR::DpdkAsmProgram *p) {
     return false;
 }
 
+const IR::Node* AddNewMetadataFields::preorder(IR::DpdkStructType *st) {
```

Comment:
Would it make sense to add a new file for this? This isn't a general optimization, rather a mandatory transformation.